src/hooks/use-accordion.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react'
+
+type AccordionState = number[] | number | null
+type SetAccordionState = (value: number) => void
+
+interface Options {
+  toggle?: boolean
+}
+
+function useAccordions(
+  initialState: number[],
+  options?: Options
+): [number[], SetAccordionState]
+
+function useAccordions(
+  initialState: number,
+  options: { toggle: false }
+): [number, SetAccordionState]
```

Comment:
Do we really need to use function overloading here? Please consider using generics instead